### PR TITLE
Small code cleanup

### PR DESCRIPTION
**Pull request by fabiocav**

### CODE REVIEW

File: src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs [hunk 1]
```diff
@@ -139,50 +139,12 @@ private JObject LoadHostConfigurationFile()
                 string sanitizedJson = SanitizeHostJson(hostConfigObject);
                 string readFileMessage = $"Host configuration file read:{NewLine}{sanitizedJson}";
 
-                // TODO: DI (FACAVAL) See method comments in ScriptHost
-                //ApplyConfiguration(hostConfigObject, ScriptConfig, _startupLogger);
-
-                // TODO: DI (FACAVAL) Review - this should likely move and not be just config
-                //if (_settingsManager.FileSystemIsReadOnly)
```
Comment (fabiocav) [inline src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs]: Moved to setup below

File: src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs [hunk 2]
```diff
@@ -139,50 +139,12 @@ private JObject LoadHostConfigurationFile()
...
-                //{
-                //    // we're in read-only mode so source files can't change
-                //    ScriptConfig.FileWatchingEnabled = false;
-                //}
-
-                // now the configuration has been read and applied re-create the logger
-                // factory and loggers ensuring that filters and settings have been applied
-                // TODO: DI (FACAVAL) TODO
-                //ConfigureLoggerFactory(recreate: true);
```
Comment (fabiocav) [inline src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs]: All logger configuration is handled by the logging changes binding to the appropriate section.

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs [hunk 3]
```diff
@@ -135,129 +142,5 @@ private void ConfigureLanguageWorkers(IConfigurationSection rootConfig, ScriptJo
...
-        // TODO: DI (FACAVAL) Moved this from ScriptHost. Validate we're applying all configuration
-        // Also noticed a lot of discrepancy between the original logic and the documentation and schema. We need to address
-        // that. The current implemenation will match the original, which means that schema and docs will need to be udpated.
-        //internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration scriptConfig, ILogger logger = null)
-        //{
-        //    var hostConfig = scriptConfig.HostOptions;
-
-        //    hostConfig.HostConfigMetadata = config;
-
-        //    JArray functions = (JArray)config["functions"];
```
Comment (fabiocav) [inline src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: Automatically handled by binding to configuration.

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs [hunk 4]
```diff
@@ -135,129 +142,5 @@ private void ConfigureLanguageWorkers(IConfigurationSection rootConfig, ScriptJo
...
-        //            scriptConfig.Functions.Add((string)function);
-        //        }
-        //    }
-        //    else
-        //    {
-        //        scriptConfig.Functions = null;
-        //    }
-
-        //    // We may already have a host id, but the one from the JSON takes precedence
-        //    JToken hostId = (JToken)config["id"];
```
Comment (fabiocav) [inline src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: No longer supported.

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs [hunk 5]
```diff
@@ -135,129 +142,5 @@ private void ConfigureLanguageWorkers(IConfigurationSection rootConfig, ScriptJo
...
-        //    if (hostId != null)
-        //    {
-        //        hostConfig.HostId = (string)hostId;
-        //    }
-
-        //    // Default AllowHostPartialStartup to true, but allow it
-        //    // to be overridden by config
-        //    hostConfig.AllowPartialHostStartup = true;
```
Comment (fabiocav) [inline src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: Automatically handled by binding to configuration.

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs [hunk 6]
```diff
@@ -135,129 +142,5 @@ private void ConfigureLanguageWorkers(IConfigurationSection rootConfig, ScriptJo
...
-        //    JToken allowPartialHostStartup = (JToken)config["allowPartialHostStartup"];
-        //    if (allowPartialHostStartup != null && allowPartialHostStartup.Type == JTokenType.Boolean)
-        //    {
-        //        hostConfig.AllowPartialHostStartup = (bool)allowPartialHostStartup;
-        //    }
-
-        //    JToken fileWatchingEnabled = (JToken)config["fileWatchingEnabled"];
```
Comment (fabiocav) [inline src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: Automatically handled by binding to configuration.

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs [hunk 7]
```diff
@@ -135,129 +142,5 @@ private void ConfigureLanguageWorkers(IConfigurationSection rootConfig, ScriptJo
...
-        //    if (fileWatchingEnabled != null && fileWatchingEnabled.Type == JTokenType.Boolean)
-        //    {
-        //        scriptConfig.FileWatchingEnabled = (bool)fileWatchingEnabled;
-        //    }
-
-        //    // Configure the set of watched directories, adding the standard built in
-        //    // set to any the user may have specified
-        //    if (scriptConfig.WatchDirectories == null)
```
Comment (fabiocav) [inline src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: Automatically handled by binding to configuration.

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs [hunk 8]
```diff
@@ -135,129 +142,5 @@ private void ConfigureLanguageWorkers(IConfigurationSection rootConfig, ScriptJo
...
-        //    JToken watchDirectories = config["watchDirectories"];
-        //    if (watchDirectories != null && watchDirectories.Type == JTokenType.Array)
-        //    {
-        //        foreach (JToken directory in watchDirectories.Where(p => p.Type == JTokenType.String))
-        //        {
-        //            scriptConfig.WatchDirectories.Add((string)directory);
-        //        }
-        //    }
-
-        //    JToken nugetFallbackFolder = config["nugetFallbackFolder"];
```
Comment (fabiocav) [inline src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: Automatically handled by binding to configuration.

File: src/WebJobs.Script/PropertyHelper.cs [hunk 9]
```diff
@@ -1,515 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
```
Comment (fabiocav) [inline src/WebJobs.Script/PropertyHelper.cs]: Dead class.

File: test/WebJobs.Script.Tests.Shared/SingleThreadedSynchronizationContext.cs [hunk 10]
```diff
@@ -1,52 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
-
-using System;
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.Shared/SingleThreadedSynchronizationContext.cs]: Dead class.

**Comment by paulbatum** [comment]
As discussed, my one question is whether any file watches on files outside of wwwroot (such as secrets) are incorrectly disabled by this.

**Comment by fabiocav** [comment]
The flag only applies to script file watching, i.e. files under the script root path (wwwroot on Azure).

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs (hunk 8 above)
Comment (mathewc) [reply src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: It still doesn't look like the fallback folder is supported, right?

Comment (fabiocav) [reply src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: It is. This will be set as part of the bind call.

**Pull request merged**